geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/package.scala

```diff
@@ -0,0 +1,44 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.spark
+
+import org.apache.spark.sql.{SQLContext, SparkSession}
+import org.locationtech.geomesa.spark.jts.encoders.SpatialEncoders
+
+/**
+ * User-facing module imports, sufficient for accessing the standard Spark-JTS functionality.
+ */
+package object jts extends DataFrameFunctions.Library with SpatialEncoders {
+  /**
+   * Initialization function that must be called before any JTS functionality
+   * is accessed. This function can be called directly, or one of the `initJTS`
+   * enrichment methods on [[SQLContext]] or [[SparkSession]] can be used instead.
+   */
+  def initJTS(sqlContext: SQLContext): Unit = {
+    org.apache.spark.sql.jts.registerTypes()
+    udf.registerFunctions(sqlContext)
+    rules.registerOptimizations(sqlContext)
+  }
+
+  /** Enrichment over [[SQLContext]] to add `withJTS` "literate" method. */
+  implicit class SQLContextWithJTS(sqlContext: SQLContext) {
```

Comment:
can we make this `implicit class SQLContextWithJTS(val sqlContext: SQLContext) extends AnyVal` - that way we make it a value class and can avoid allocating a new class instance: https://docs.scala-lang.org/overviews/core/value-classes.html

Reply:
+1